diff -up sage-5.10/spkg/build/sage-5.10/module_list.py.orig sage-5.10/spkg/build/sage-5.10/module_list.py
--- sage-5.10/spkg/build/sage-5.10/module_list.py.orig	2013-07-29 20:09:16.498087472 -0300
+++ sage-5.10/spkg/build/sage-5.10/module_list.py	2013-07-29 20:09:30.624088013 -0300
@@ -2017,7 +2017,7 @@ if 1:
         )
 
 
-if is_package_installed('cryptominisat'):
+if 1:
     ext_modules.extend([
         Extension("sage.sat.solvers.cryptominisat.cryptominisat",
                   ["sage/sat/solvers/cryptominisat/cryptominisat.pyx"],
