diff -up sage-5.10/spkg/build/sage-5.10/sage/ext/cdefs.pxi.orig sage-5.10/spkg/build/sage-5.10/sage/ext/cdefs.pxi
--- sage-5.10/spkg/build/sage-5.10/sage/ext/cdefs.pxi.orig	2013-07-29 20:12:52.183095731 -0300
+++ sage-5.10/spkg/build/sage-5.10/sage/ext/cdefs.pxi	2013-07-29 20:13:06.111096264 -0300
@@ -5,6 +5,9 @@
 from libc.stdio cimport *
 from libc.string cimport strlen, strcpy, memset, memcpy
 
-from libc.math cimport sqrt, frexp, ldexp
+cdef extern from "math.h":
+    cdef double sqrt "sqrt" (double)
+    cdef double frexp "frexp" (double, int*)
+    cdef double ldexp "ldexp" (double, int)
 
 from sage.libs.gmp.all cimport *
