diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 12:46:14.237860892 -0200
+++ src/module_list.py	2014-02-07 12:46:18.226861045 -0200
@@ -2073,7 +2073,7 @@ ext_modules = [
 def is_package_installed(name):
     return False
 
-if is_package_installed('fes'):
+if 1:
     ext_modules.extend([
        Extension("sage.libs.fes",
                  ["sage/libs/fes.pyx"],
