--- sage-5.4.beta1/spkg/build/sage-5.4.beta1/sage/interfaces/lie.py.orig	2012-11-06 13:58:05.290358692 -0200
+++ sage-5.4.beta1/spkg/build/sage-5.4.beta1/sage/interfaces/lie.py	2012-11-06 13:59:12.601361648 -0200
@@ -290,6 +290,7 @@ from sage.misc.misc import verbose, DOT_
 
 COMMANDS_CACHE = '%s/lie_commandlist_cache.sobj'%DOT_SAGE
 HELP_CACHE = '%s/lie_helpdict_cache.sobj'%DOT_SAGE
+LIE_LOCAL = '/usr/share/lie'
 
 class LiE(Expect):
     r"""
@@ -322,7 +323,7 @@ class LiE(Expect):
                         prompt = '> ',
 
                         # This is the command that starts up your program
-                        command = "bash "+ SAGE_LOCAL + "/bin/lie",
+                        command = "lie",
 
                         maxread = maxread,
                         server=server,
@@ -387,7 +388,7 @@ class LiE(Expect):
 
 
         for f in filenames:
-            filename = SAGE_LOCAL + "/lib/LiE/" + f
+            filename = LIE_LOCAL + f
             info = open(filename)
             prev_command = ""
             help_text = ""
@@ -894,7 +895,7 @@ def lie_version():
         sage: lie_version() # optional - lie
         '2.1'
     """
-    f = open(SAGE_LOCAL + 'lib/LiE/INFO.0')
+    f = open(LIE_LOCAL + 'INFO.0')
     lines = f.readlines()
     f.close()
     i = lines.index('@version()\n')
