diff -up sage-5.12/spkg/build/sage-5.12/module_list.py.orig sage-5.12/spkg/build/sage-5.12/module_list.py
--- sage-5.12/spkg/build/sage-5.12/module_list.py.orig	2013-10-16 12:39:05.925265588 -0300
+++ sage-5.12/spkg/build/sage-5.12/module_list.py	2013-10-16 12:39:14.072265900 -0300
@@ -2084,7 +2084,7 @@ if is_package_installed('cbc'):
         )
 
 
-if is_package_installed('cryptominisat'):
+if 1:
     ext_modules.extend([
         Extension("sage.sat.solvers.cryptominisat.cryptominisat",
                   ["sage/sat/solvers/cryptominisat/cryptominisat.pyx"],
