diff -up sage-5.10/spkg/build/sage-5.10/module_list.py.orig sage-5.10/spkg/build/sage-5.10/module_list.py
--- sage-5.10/spkg/build/sage-5.10/module_list.py.orig	2013-07-29 20:05:39.844079175 -0300
+++ sage-5.10/spkg/build/sage-5.10/module_list.py	2013-07-29 20:05:59.925079944 -0300
@@ -2007,7 +2007,7 @@ if (os.path.isfile(SAGE_INC + "/cplex.h"
                   libraries = ["csage", "stdc++", "cplex"]) 
         )
 
-if is_package_installed('cbc'):
+if 1:
     ext_modules.append(
         Extension("sage.numerical.backends.coin_backend",                                                                                                                                                                                     
                   ["sage/numerical/backends/coin_backend.pyx"],
