diff -up src/sage/graphs/graph_generators.py.orig src/sage/graphs/graph_generators.py
--- src/sage/graphs/graph_generators.py.orig	2016-08-08 13:38:51.824541870 -0400
+++ src/sage/graphs/graph_generators.py	2016-08-08 13:39:20.460542967 -0400
@@ -872,7 +872,7 @@ class GraphGenerators():
             >A geng -d0D3 n=4 e=0-6
         """
         import subprocess
-        sp = subprocess.Popen("geng {0}".format(options), shell=True,
+        sp = subprocess.Popen("nauty-geng {0}".format(options), shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
                               stderr=subprocess.PIPE, close_fds=True)
         if debug:
