diff -up sage-5.13/spkg/build/sage-5.13/module_list.py.orig sage-5.13/spkg/build/sage-5.13/module_list.py
--- sage-5.13/spkg/build/sage-5.13/module_list.py.orig	2014-01-28 16:14:01.061216627 -0200
+++ sage-5.13/spkg/build/sage-5.13/module_list.py	2014-01-28 16:14:03.961216738 -0200
@@ -2056,7 +2056,7 @@ ext_modules = [
 def is_package_installed(name):
     return False
 
-if is_package_installed('fes'):
+if 1:
     ext_modules.extend([
        Extension("sage.libs.fes",
                  ["sage/libs/fes.pyx"],
