diff -up sage-5.13/spkg/build/sage-5.13/sage/graphs/graph_generators.py.orig sage-5.13/spkg/build/sage-5.13/sage/graphs/graph_generators.py
--- sage-5.13/spkg/build/sage-5.13/sage/graphs/graph_generators.py.orig	2014-01-28 16:10:40.018208928 -0200
+++ sage-5.13/spkg/build/sage-5.13/sage/graphs/graph_generators.py	2014-01-28 16:10:49.569209294 -0200
@@ -810,7 +810,7 @@ class GraphGenerators():
         """
         import subprocess
         from sage.misc.package import is_package_installed
-        if not is_package_installed("nauty"):
+        if 1:
             raise TypeError("the optional nauty package is not installed")
         sp = subprocess.Popen("nauty-geng {0}".format(options), shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
