--- src/module_list.py.orig	2014-05-10 15:59:50.979138032 -0300
+++ src/module_list.py	2014-05-10 16:00:00.142137375 -0300
@@ -2125,7 +2125,7 @@ if is_package_installed('cbc'):
         )
 
 
-if is_package_installed('cryptominisat'):
+if 1:
     ext_modules.extend([
         Extension("sage.sat.solvers.cryptominisat.cryptominisat",
                   ["sage/sat/solvers/cryptominisat/cryptominisat.pyx"],
