diff -up sage-5.12/spkg/build/sage-5.12/module_list.py.orig sage-5.12/spkg/build/sage-5.12/module_list.py
--- sage-5.12/spkg/build/sage-5.12/module_list.py.orig	2013-10-16 12:13:30.643206797 -0300
+++ sage-5.12/spkg/build/sage-5.12/module_list.py	2013-10-16 12:13:41.295207205 -0300
@@ -2033,7 +2033,7 @@ ext_modules = [
 def is_package_installed(name):
     return False
 
-if is_package_installed('fes'):
+if 1:
     ext_modules.extend([
        Extension("sage.libs.fes",
                  ["sage/libs/fes.pyx"],
