diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 12:38:18.873842689 -0200
+++ src/module_list.py	2014-02-07 12:38:27.546843021 -0200
@@ -2149,7 +2149,7 @@ if UNAME[0] == "Darwin" and not UNAME[2]
         )
 
 
-if is_package_installed('lrcalc'):
+if 1:
     ext_modules.append(
         Extension('sage.libs.lrcalc.lrcalc',
                   sources = ["sage/libs/lrcalc/lrcalc.pyx"],
