diff -up sage-5.10/spkg/build/sage-5.10/sage/geometry/polyhedron/base.py.orig sage-5.10/spkg/build/sage-5.10/sage/geometry/polyhedron/base.py
--- sage-5.10/spkg/build/sage-5.10/sage/geometry/polyhedron/base.py.orig	2013-05-06 14:12:21.664799886 -0300
+++ sage-5.10/spkg/build/sage-5.10/sage/geometry/polyhedron/base.py	2013-05-06 14:12:28.625800152 -0300
@@ -3281,11 +3281,6 @@ class Polyhedron_base(Element):
 
              David Avis's lrs program.
         """
-        if is_package_installed('lrs') != True:
-            print 'You must install the optional lrs package ' \
-                  'for this function to work'
-            raise NotImplementedError
-
         from sage.misc.temporary_file import tmp_filename
         from subprocess import Popen, PIPE
         in_str = self.cdd_Vrepresentation()
