diff -up src/sage/graphs/graph_generators.py.orig src/sage/graphs/graph_generators.py
--- src/sage/graphs/graph_generators.py.orig	2014-02-07 12:36:49.882839281 -0200
+++ src/sage/graphs/graph_generators.py	2014-02-07 12:37:01.793839737 -0200
@@ -823,7 +823,7 @@ class GraphGenerators():
         """
         import subprocess
         from sage.misc.package import is_package_installed
-        if not is_package_installed("nauty"):
+        if 1:
             raise TypeError("the optional nauty package is not installed")
         sp = subprocess.Popen("nauty-geng {0}".format(options), shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
