--- sage-5.12/spkg/build/sage-5.12/c_lib/src/memory.c.orig	2012-07-10 11:46:54.516595967 -0400
+++ sage-5.12/spkg/build/sage-5.12/c_lib/src/memory.c	2012-07-10 11:47:12.688596755 -0400
@@ -17,7 +17,7 @@ AUTHORS:
  *                  http://www.gnu.org/licenses/
  ****************************************************************************/
 
-#include <mpir.h>
+#include <gmp.h>
 #include "memory.h"
 
 /* mpir memory functions */
