diff -up sage-5.5/spkg/build/sage-5.5/sage/sandpiles/sandpile.py.orig sage-5.5/spkg/build/sage-5.5/sage/sandpiles/sandpile.py
--- sage-5.5/spkg/build/sage-5.5/sage/sandpiles/sandpile.py.orig	2012-07-11 14:39:38.474386100 -0400
+++ sage-5.5/spkg/build/sage-5.5/sage/sandpiles/sandpile.py	2012-07-11 14:40:33.870388225 -0400
@@ -22,9 +22,6 @@ http://sagemath.org/download-packages.ht
 packages.  An alternative is to install 4ti2 separately, then point the
 following variable to the correct path.
 """
-SAGE_ROOT = os.environ['SAGE_ROOT']
-path_to_zsolve = SAGE_ROOT+'/local/bin/'
-
 r"""
 Sage Sandpiles
 
@@ -4097,7 +4094,7 @@ class SandpileDivisor(dict):
         sign_file.close()
         # compute
         try:
-            os.system(path_to_zsolve+'zsolve -q ' + lin_sys + ' > ' + lin_sys_log)
+            os.system('zsolve -q ' + lin_sys + ' > ' + lin_sys_log)
             # process the results
             zhom_file = open(lin_sys_zhom,'r')
         except IOError:
