diff -up sage-5.13/spkg/build/sage-5.13/sage/sandpiles/sandpile.py.orig sage-5.13/spkg/build/sage-5.13/sage/sandpiles/sandpile.py
--- sage-5.13/spkg/build/sage-5.13/sage/sandpiles/sandpile.py.orig	2014-01-28 16:01:55.841188856 -0200
+++ sage-5.13/spkg/build/sage-5.13/sage/sandpiles/sandpile.py	2014-01-28 16:02:07.914189318 -0200
@@ -24,8 +24,6 @@ packages.  An alternative is to install
 following variable to the correct path.
 """
 
-path_to_zsolve = os.path.join(SAGE_LOCAL,'bin','zsolve')
-
 r"""
 Sage Sandpiles
 
@@ -4097,7 +4095,7 @@ class SandpileDivisor(dict):
         sign_file.close()
         # compute
         try:
-            os.system(path_to_zsolve+' -q ' + lin_sys + ' > ' + lin_sys_log)
+            os.system('zsolve -q ' + lin_sys + ' > ' + lin_sys_log)
             # process the results
             zhom_file = open(lin_sys_zhom,'r')
         except IOError:
