--- src/module_list.py.orig	2014-05-10 16:00:52.758133601 -0300
+++ src/module_list.py	2014-05-10 16:01:04.088132788 -0300
@@ -1860,7 +1860,7 @@ ext_modules = [
     Extension('sage.rings.polynomial.pbori',
               sources = ['sage/rings/polynomial/pbori.pyx'],
               libraries=['polybori-' + polybori_major_version,
-                         'polybori_groebner-' + polybori_major_version, 'm4ri', 'png'],
+                         'polybori_groebner-' + polybori_major_version, 'cudd', 'm4ri', 'png'],
               include_dirs = [SAGE_INC, "sage/libs/polybori"],
               depends = [SAGE_INC + "/polybori/" + hd + ".h" for hd in ["polybori", "config"] ] + \
                         [SAGE_INC + '/m4ri/m4ri.h'],
