diff -up src/sage/geometry/polyhedron/base.py.orig src/sage/geometry/polyhedron/base.py
--- src/sage/geometry/polyhedron/base.py.orig	2014-02-07 12:36:06.514837620 -0200
+++ src/sage/geometry/polyhedron/base.py	2014-02-07 12:36:16.918838019 -0200
@@ -3406,11 +3406,6 @@ class Polyhedron_base(Element):
 
              David Avis's lrs program.
         """
-        if is_package_installed('lrs') != True:
-            print 'You must install the optional lrs package ' \
-                  'for this function to work'
-            raise NotImplementedError
-
         from sage.misc.temporary_file import tmp_filename
         from subprocess import Popen, PIPE
         in_str = self.cdd_Vrepresentation()
