diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 11:50:03.665731821 -0200
+++ src/module_list.py	2014-02-07 11:50:13.454732196 -0200
@@ -1066,7 +1066,7 @@ ext_modules = [
 
     Extension('sage.matrix.matrix_mod2_dense',
               sources = ['sage/matrix/matrix_mod2_dense.pyx'],
-              libraries = ['gmp','m4ri', 'gd', 'png12', 'z'],
+              libraries = ['gmp','m4ri', 'gd', 'png', 'z'],
               extra_compile_args = m4ri_extra_compile_args,
               depends = [SAGE_INC + "/png.h", SAGE_INC + "/m4ri/m4ri.h"]),
 
@@ -1369,7 +1369,7 @@ ext_modules = [
 
     Extension('sage.modules.vector_mod2_dense',
               sources = ['sage/modules/vector_mod2_dense.pyx'],
-              libraries = ['gmp','m4ri', 'png12', 'gd'],
+              libraries = ['gmp','m4ri', 'png', 'gd'],
               extra_compile_args = m4ri_extra_compile_args,
               depends = [SAGE_INC + "/png.h", SAGE_INC + "/m4ri/m4ri.h"]),
 
@@ -1859,7 +1859,7 @@ ext_modules = [
     Extension('sage.rings.polynomial.pbori',
               sources = ['sage/rings/polynomial/pbori.pyx'],
               libraries=['polybori-' + polybori_major_version,
-                         'polybori_groebner-' + polybori_major_version, 'm4ri', 'png12'],
+                         'polybori_groebner-' + polybori_major_version, 'm4ri', 'png'],
               include_dirs = [SAGE_INC, "sage/libs/polybori"],
               depends = [SAGE_INC + "/polybori/" + hd + ".h" for hd in ["polybori", "config"] ] + \
                         [SAGE_INC + '/m4ri/m4ri.h'],
