diff -up sage-5.13/spkg/build/sage-5.13/module_list.py.orig sage-5.13/spkg/build/sage-5.13/module_list.py
--- sage-5.13/spkg/build/sage-5.13/module_list.py.orig	2014-01-28 16:20:58.178232600 -0200
+++ sage-5.13/spkg/build/sage-5.13/module_list.py	2014-01-28 16:21:11.779233120 -0200
@@ -2107,7 +2107,7 @@ if is_package_installed('cbc'):
         )
 
 
-if is_package_installed('cryptominisat'):
+if 1:
     ext_modules.extend([
         Extension("sage.sat.solvers.cryptominisat.cryptominisat",
                   ["sage/sat/solvers/cryptominisat/cryptominisat.pyx"],
