diff -up sage-5.5/spkg/build/sage-5.5/sage/databases/cremona.py.orig sage-5.5/spkg/build/sage-5.5/sage/databases/cremona.py
--- sage-5.5/spkg/build/sage-5.5/sage/databases/cremona.py.orig	2013-01-19 11:44:58.950163972 -0200
+++ sage-5.5/spkg/build/sage-5.5/sage/databases/cremona.py	2013-01-19 11:45:57.644166220 -0200
@@ -889,14 +889,9 @@ class MiniCremonaDatabase(SQLDatabase):
             if N < self.largest_conductor():
                 message = "There is no elliptic curve with label " + label \
                     + " in the database"
-            elif is_package_installed('database_cremona_ellcurve'):
-                message = "There is no elliptic curve with label " + label \
-                    + " in the currently available databases"
             else:
                 message = "There is no elliptic curve with label " \
-                    + label + " in the default database; try installing " \
-                    + "the optional package database_cremona_ellcurve which " \
-                    + "contains the complete Cremona database"
+                    + label + " in the default database"
             raise ValueError(message)
 
     def iter(self, conductors):
@@ -1598,10 +1593,12 @@ def CremonaDatabase(name=None,mini=None,
     if name is None and not set_global:
         return _db
     if set_global and name is None:
-        if is_package_installed('database_cremona_ellcurve'):
-            name = 'cremona'
-        else:
-            name = 'cremona mini'
+        # currently the sagemath rpm package only installs cremona mini
+        #if is_package_installed('database_cremona_ellcurve'):
+        #    name = 'cremona'
+        #else:
+        #    name = 'cremona mini'
+        name = 'cremona mini'
     if name == 'cremona':
         mini = False
     elif name == 'cremona mini':
