diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 12:39:09.255844618 -0200
+++ src/module_list.py	2014-02-07 12:39:20.005845030 -0200
@@ -2114,7 +2114,7 @@ if (os.path.isfile(SAGE_INC + "/cplex.h"
                   libraries = ["csage", "stdc++", "cplex"])
         )
 
-if is_package_installed('cbc'):
+if 1:
     ext_modules.append(
         Extension("sage.numerical.backends.coin_backend",
                   ["sage/numerical/backends/coin_backend.pyx"],
