diff -up src/sage/all.py.orig src/sage/all.py
--- src/sage/all.py.orig	2014-02-07 12:43:35.244854804 -0200
+++ src/sage/all.py	2014-02-07 12:44:11.798856204 -0200
@@ -325,11 +325,11 @@ def _write_started_file():
 
     Check that the file exists when Sage is running::
 
-        sage: started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+        sage: started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
         sage: os.path.isfile(started_file)
         True
     """
-    started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+    started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
     # Do nothing if the file already exists
     if os.path.isfile(started_file):
         return
