--- sage-5.4.beta1/spkg/build/sage-5.4.beta1/sage/interfaces/gp.py.orig	2011-11-04 23:32:52.523025905 -0200
+++ sage-5.4.beta1/spkg/build/sage-5.4.beta1/sage/interfaces/gp.py	2011-11-04 23:33:09.624117731 -0200
@@ -981,7 +981,7 @@ from sage.misc.all import DOT_SAGE, SAGE
 import os
 
 # Set GPRC environment variable to $SAGE_ROOT/local/etc/gprc.expect
-os.environ["GPRC"] = '%s/local/etc/gprc.expect'%SAGE_ROOT
+os.environ["GPRC"] = '%s/data/extcode/pari/gprc.expect'%SAGE_ROOT
 
 # An instance
 gp = Gp(logfile=DOT_SAGE+'/gp-expect.log') # useful for debugging!
