--- sage-5.4.1/spkg/build/networkx-1.6/src/networkx/convert.py.orig	2012-08-20 10:52:06.850123089 -0300
+++ sage-5.4.1/spkg/build/networkx-1.6/src/networkx/convert.py	2012-08-20 10:53:38.184120582 -0300
@@ -150,13 +150,27 @@ def to_networkx_graph(data,create_using=
     # numpy matrix or ndarray 
     try:
         import numpy
-        if isinstance(data,numpy.matrix) or \
+        #numpy.core.defmatrix changed to numpy.matrixlib.defmatrix 
+        #in the versions 1.3.x to 1.4.x 
+	
+        try: 
+        #numpy-1.4.1 
+            if isinstance(data,numpy.matrixlib.defmatrix.matrix) or \
                isinstance(data,numpy.ndarray):
-            try:
-                return from_numpy_matrix(data,create_using=create_using)
-            except:
-                raise nx.NetworkXError(\
-                  "Input is not a correct numpy matrix or array.")
+                try:
+                    return from_numpy_matrix(data,create_using=create_using)
+	        except:
+                    raise networkx.NetworkXError,\
+                        "Input is not a correct numpy matrix or array."
+	except AttributeError:
+	    #numpy-1.3.x or less
+	    if isinstance(data,numpy.core.defmatrix.matrix) or \
+	       isinstance(data,numpy.ndarray):
+		try:
+		    return from_numpy_matrix(data,create_using=create_using)
+		except:
+		    raise networkx.NetworkXError,\
+			"Input is not a correct numpy matrix or array."
     except ImportError:
         warnings.warn('numpy not found, skipping conversion test.',
                       ImportWarning)
--- sage-5.4.1/spkg/build/sage-5.4.1/sage/matrix/matrix1.pyx.orig	2012-08-20 10:54:52.248118441 -0300
+++ sage-5.4.1/spkg/build/sage-5.4.1/sage/matrix/matrix1.pyx	2012-08-20 10:54:54.774118663 -0300
@@ -492,7 +492,7 @@ cdef class Matrix(matrix0.Matrix):
 
             sage: import numpy
             sage: sorted(numpy.typecodes.items())
-            [('All', '?bhilqpBHILQPfdgFDGSUVO'), ('AllFloat', 'fdgFDG'), ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'), ('Float', 'fdg'), ('Integer', 'bhilqp'), ('UnsignedInteger', 'BHILQP')]
+            [('All', '?bhilqpBHILQPfdgFDGSUVOMm'), ('AllFloat', 'fdgFDG'), ('AllInteger', 'bBhHiIlLqQpP'), ('Character', 'c'), ('Complex', 'FDG'), ('Datetime', 'Mm'), ('Float', 'fdg'), ('Integer', 'bhilqp'), ('UnsignedInteger', 'BHILQP')]
 
         Alternatively, numpy automatically calls this function (via
         the magic :meth:`__array__` method) to convert Sage matrices
