diff -up sage-5.13/spkg/build/sage-5.13/module_list.py.orig sage-5.13/spkg/build/sage-5.13/module_list.py
--- sage-5.13/spkg/build/sage-5.13/module_list.py.orig	2014-01-28 16:12:03.635212130 -0200
+++ sage-5.13/spkg/build/sage-5.13/module_list.py	2014-01-28 16:12:12.674212476 -0200
@@ -2132,7 +2132,7 @@ if UNAME[0] == "Darwin" and not UNAME[2]
         )
 
 
-if is_package_installed('lrcalc'):
+if 1:
     ext_modules.append(
         Extension('sage.libs.lrcalc.lrcalc',
                   sources = ["sage/libs/lrcalc/lrcalc.pyx"],
