--- src/sage/libs/pari/gen.pyx.orig	2014-04-20 21:13:45.990848860 -0300
+++ src/sage/libs/pari/gen.pyx	2014-04-20 21:14:56.465845890 -0300
@@ -6932,11 +6932,11 @@ cdef class gen(sage.structure.element.Ri
         cdef GEN disc
         if not fa:
             pari_catch_sig_on()
-            B = P.new_gen_noclear(nfbasis(self.g, &disc, flag, NULL))
+            B = P.new_gen_noclear(nfbasis(self.g, &disc, NULL))
         else:
             t0 = objtogen(fa)
             pari_catch_sig_on()
-            B = P.new_gen_noclear(nfbasis(self.g, &disc, flag, t0.g))
+            B = P.new_gen_noclear(nfbasis(self.g, &disc, t0.g))
         D = P.new_gen(disc)
         return B, D
 
--- src/sage/libs/pari/decl.pxi.orig	2014-04-20 22:28:36.551659642 -0300
+++ src/sage/libs/pari/decl.pxi	2014-04-20 22:28:47.624659175 -0300
@@ -724,7 +724,7 @@ cdef extern from 'pari/pari.h':
     long    idealval(GEN nf,GEN ix,GEN vp)
     GEN     idealprodprime(GEN nf, GEN L)
     GEN     indexpartial(GEN P, GEN DP)
-    GEN     nfbasis(GEN x, GEN *y,long flag,GEN p)
+    GEN     nfbasis(GEN x, GEN *y,GEN p)
     GEN     nfbasis0(GEN x,long flag,GEN p)
     GEN     nfdisc0(GEN x,long flag, GEN p)
     GEN     nfreducemodpr(GEN nf, GEN x, GEN modpr)
