--- src/sage/all.py.orig	2014-05-10 15:45:46.934198573 -0300
+++ src/sage/all.py	2014-05-10 15:45:58.395197751 -0300
@@ -327,11 +327,11 @@ def _write_started_file():
 
     Check that the file exists when Sage is running::
 
-        sage: started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+        sage: started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
         sage: os.path.isfile(started_file)
         True
     """
-    started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+    started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
     # Do nothing if the file already exists
     if os.path.isfile(started_file):
         return
