--- src/module_list.py.orig	2014-05-10 16:02:43.371125667 -0300
+++ src/module_list.py	2014-05-10 16:02:58.362124591 -0300
@@ -2115,7 +2115,7 @@ if (os.path.isfile(SAGE_INC + "/cplex.h"
                   libraries = ["stdc++", "cplex"])
         )
 
-if is_package_installed('cbc'):
+if 1:
     ext_modules.append(
         Extension("sage.numerical.backends.coin_backend",
                   ["sage/numerical/backends/coin_backend.pyx"],
