diff -up sage-5.12/spkg/build/sage-5.12/sage/graphs/graph_generators.py.orig sage-5.12/spkg/build/sage-5.12/sage/graphs/graph_generators.py
--- sage-5.12/spkg/build/sage-5.12/sage/graphs/graph_generators.py.orig	2013-10-16 12:18:25.218218077 -0300
+++ sage-5.12/spkg/build/sage-5.12/sage/graphs/graph_generators.py	2013-10-16 12:18:46.709218900 -0300
@@ -793,7 +793,7 @@ class GraphGenerators():
         """
         import subprocess
         from sage.misc.package import is_package_installed
-        if not is_package_installed("nauty"):
+        if 1:
             raise TypeError("the optional nauty package is not installed")
         sp = subprocess.Popen("nauty-geng {0}".format(options), shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
