--- a/src/setup.py.o      2016-12-27 19:32:19.661466958 +0000
+++ a/src/setup.py        2016-12-27 19:32:40.819715908 +0000
@@ -564,9 +564,9 @@
 
     # Sage uses these directives (mostly for historical reasons).
     Cython.Compiler.Options.embed_pos_in_docstring = True
-    Cython.Compiler.Options.directive_defaults['autotestdict'] = False
-    Cython.Compiler.Options.directive_defaults['cdivision'] = True
-    Cython.Compiler.Options.directive_defaults['fast_getattr'] = True
+    Cython.Compiler.Options._directive_defaults['autotestdict'] = False
+    Cython.Compiler.Options._directive_defaults['cdivision'] = True
+    Cython.Compiler.Options._directive_defaults['fast_getattr'] = True
     # The globals() builtin in Cython was fixed to return to the current scope,
     # but Sage relies on the broken behavior of returning to the nearest
     # enclosing Python scope (e.g. to perform variable injection).
