diff -up src/sage/libs/gap/test/main.c.orig src/sage/libs/gap/test/main.c
--- src/sage/libs/gap/test/main.c.orig	2016-08-08 13:43:32.376552613 -0400
+++ src/sage/libs/gap/test/main.c	2016-08-08 13:43:38.209552837 -0400
@@ -22,7 +22,7 @@ void eval(char* cmd) {
   libgap_start_interaction(cmd);
 
   libgap_enter();
-  libGAP_ReadEvalCommand(libGAP_BottomLVars);
+  libGAP_ReadEvalCommand(libGAP_BottomLVars, NULL);
   libGAP_ViewObjHandler(libGAP_ReadEvalResult);
   char* out = libgap_get_output();
   libgap_exit();
diff -up src/sage/libs/gap/util.pyx.orig src/sage/libs/gap/util.pyx
--- src/sage/libs/gap/util.pyx.orig	2016-08-08 13:41:46.176548547 -0400
+++ src/sage/libs/gap/util.pyx	2016-08-08 13:42:26.594550094 -0400
@@ -154,18 +154,9 @@ def gap_root():
 
         sage: from sage.libs.gap.util import gap_root
         sage: gap_root()   # random output
-        '/home/vbraun/opt/sage-5.3.rc0/local/gap/latest'
+        '/usr/lib/gap'
     """
-    import os.path
-    gapdir = os.path.join(SAGE_LOCAL, 'gap', 'latest')
-    if os.path.exists(gapdir):
-        return gapdir
-    print('The gap-4.5.5.spkg (or later) seems to be not installed!')
-    gap_sh = open(os.path.join(SAGE_LOCAL, 'bin', 'gap')).read().splitlines()
-    gapdir = filter(lambda dir:dir.strip().startswith('GAP_DIR'), gap_sh)[0]
-    gapdir = gapdir.split('"')[1]
-    gapdir = gapdir.replace('$SAGE_LOCAL', SAGE_LOCAL)
-    return gapdir
+    return '/usr/lib/gap'
 
 
 cdef initialize():
