diff -up sage-5.13/spkg/build/sage-5.13/module_list.py.orig sage-5.13/spkg/build/sage-5.13/module_list.py
--- sage-5.13/spkg/build/sage-5.13/module_list.py.orig	2014-01-28 16:12:52.010213983 -0200
+++ sage-5.13/spkg/build/sage-5.13/module_list.py	2014-01-28 16:13:00.749214317 -0200
@@ -2097,7 +2097,7 @@ if (os.path.isfile(SAGE_INC + "/cplex.h"
                   libraries = ["csage", "stdc++", "cplex"]) 
         )
 
-if is_package_installed('cbc'):
+if 1:
     ext_modules.append(
         Extension("sage.numerical.backends.coin_backend",                                                                                                                                                                                     
                   ["sage/numerical/backends/coin_backend.pyx"],
