diff -up src/sage/libs/pari/decl.pxi.orig src/sage/libs/pari/decl.pxi
--- src/sage/libs/pari/decl.pxi.orig	2014-02-07 12:47:51.412864613 -0200
+++ src/sage/libs/pari/decl.pxi	2014-02-07 12:48:00.616864966 -0200
@@ -869,7 +869,6 @@ cdef extern from 'pari/pari.h':
     GEN     polredabs0(GEN x, long flag)
     GEN     polredabs2(GEN x)
     GEN     polredabsall(GEN x, long flun)
-    GEN     polredbest(GEN x, long flag)
     GEN     qflll0(GEN x, long flag)
     GEN     qflllgram0(GEN x, long flag)
     GEN     smallpolred(GEN x)
diff -up src/sage/libs/pari/gen.pyx.orig src/sage/libs/pari/gen.pyx
--- src/sage/libs/pari/gen.pyx.orig	2014-02-07 12:48:07.252865220 -0200
+++ src/sage/libs/pari/gen.pyx	2014-02-07 12:48:21.513865766 -0200
@@ -7885,10 +7885,6 @@ cdef class gen(sage.structure.element.Ri
         pari_catch_sig_on()
         return P.new_gen(polredabs0(self.g, flag))
 
-    def polredbest(self, long flag=0):
-        pari_catch_sig_on()
-        return P.new_gen(polredbest(self.g, flag))
-
     def polresultant(self, y, var=-1, long flag=0):
         cdef gen t0 = objtogen(y)
         pari_catch_sig_on()
