diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 12:47:06.312862886 -0200
+++ src/module_list.py	2014-02-07 12:49:07.920867543 -0200
@@ -2124,7 +2124,7 @@ if is_package_installed('cbc'):
         )
 
 
-if is_package_installed('cryptominisat'):
+if 1:
     ext_modules.extend([
         Extension("sage.sat.solvers.cryptominisat.cryptominisat",
                   ["sage/sat/solvers/cryptominisat/cryptominisat.pyx"],
