diff -up sage-5.10/spkg/build/sage-5.10/sage/sandpiles/sandpile.py.orig sage-5.10/spkg/build/sage-5.10/sage/sandpiles/sandpile.py
--- sage-5.10/spkg/build/sage-5.10/sage/sandpiles/sandpile.py.orig	2013-05-06 13:50:14.882749079 -0300
+++ sage-5.10/spkg/build/sage-5.10/sage/sandpiles/sandpile.py	2013-05-06 13:50:34.526749831 -0300
@@ -24,8 +24,6 @@ packages.  An alternative is to install
 following variable to the correct path.
 """
 
-path_to_zsolve = os.path.join(SAGE_LOCAL,'bin','zsolve')
-
 r"""
 Sage Sandpiles
 
@@ -4098,7 +4096,7 @@ class SandpileDivisor(dict):
         sign_file.close()
         # compute
         try:
-            os.system(path_to_zsolve+' -q ' + lin_sys + ' > ' + lin_sys_log)
+            os.system('zsolve -q ' + lin_sys + ' > ' + lin_sys_log)
             # process the results
             zhom_file = open(lin_sys_zhom,'r')
         except IOError:
