diff -up sage-5.13/spkg/build/sage-5.13/c_lib/src/memory.c.orig sage-5.13/spkg/build/sage-5.13/c_lib/src/memory.c
--- sage-5.13/spkg/build/sage-5.13/c_lib/src/memory.c.orig	2014-01-28 15:45:05.920150182 -0200
+++ sage-5.13/spkg/build/sage-5.13/c_lib/src/memory.c	2014-01-28 15:45:14.225150500 -0200
@@ -18,7 +18,7 @@ AUTHORS:
  ****************************************************************************/
 
 #include <Python.h>
-#include <mpir.h>
+#include <gmp.h>
 #include "interrupt.h"
 #include "memory.h"
 
