--- src/sage/graphs/graph_generators.py.orig	2014-05-10 15:51:24.250174378 -0300
+++ src/sage/graphs/graph_generators.py	2014-05-10 15:51:36.224173519 -0300
@@ -824,7 +824,7 @@ class GraphGenerators():
         """
         import subprocess
         from sage.misc.package import is_package_installed
-        if not is_package_installed("nauty"):
+        if 1:
             raise TypeError("the optional nauty package is not installed")
         sp = subprocess.Popen("nauty-geng {0}".format(options), shell=True,
                               stdin=subprocess.PIPE, stdout=subprocess.PIPE,
