diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2016-08-08 13:18:16.519494566 -0400
+++ src/module_list.py	2016-08-08 13:18:29.570495066 -0400
@@ -3,7 +3,7 @@ from glob import glob
 from distutils.extension import Extension
 from sage.env import SAGE_LOCAL
 
-SAGE_INC = os.path.join(SAGE_LOCAL, 'include')
+SAGE_INC = '/usr/include'
 
 #########################################################
 ### pkg-config setup
diff -up src/setup.py.orig src/setup.py
--- src/setup.py.orig	2016-08-08 13:19:09.383496590 -0400
+++ src/setup.py	2016-08-08 13:20:41.161500105 -0400
@@ -561,11 +561,9 @@ def run_cythonize():
         print('Enabling Cython profiling support')
         profile = True
 
-    # Disable Cython caching (the cache is stored in ~/.cycache which is
+    # Enable Cython caching (the cache is stored in ~/.cycache which is
     # Cython's default).
-    # Cycache is currently too broken to use reliably:
-    # http://trac.sagemath.org/ticket/17851
-    Cython.Compiler.Main.default_options['cache'] = False
+    Cython.Compiler.Main.default_options['cache'] = True
 
     force = True
     version_file = os.path.join(SAGE_CYTHONIZED, '.cython_version')
@@ -618,21 +616,6 @@ print("Discovered Python/Cython sources,
 
 
 #########################################################
-### Clean
-#########################################################
-
-print('Cleaning up stale installed files....')
-t = time.time()
-from sage_setup.clean import clean_install_dir
-output_dirs = SITE_PACKAGES + glob.glob(os.path.join(build_base, 'lib*'))
-for output_dir in output_dirs:
-    print('- cleaning {0}'.format(output_dir))
-    clean_install_dir(output_dir, python_packages, python_modules,
-            ext_modules, python_data_files)
-print('Finished cleaning, time: %.2f seconds.' % (time.time() - t))
-
-
-#########################################################
 ### Install also Jupyter kernel spec
 #########################################################
 
diff -up src/sage/misc/cython.py.orig src/sage/misc/cython.py
--- src/sage/misc/cython.py.orig	2016-08-24 11:51:48.802265784 -0400
+++ src/sage/misc/cython.py	2016-08-24 11:52:36.723267619 -0400
@@ -426,7 +426,7 @@ extra_compile_args = %s
 
 ext_modules = [Extension('%s', sources=['%s.%s', %s],
                      libraries=%s,
-                     library_dirs=[SAGE_LOCAL + '/lib/'] + %s,
+                     library_dirs=[os.path.realpath(SAGE_LOCAL + '/lib/')] + %s,
                      extra_compile_args = extra_compile_args,
                      language = '%s' )]
 
