diff -up sage-5.12/spkg/build/sage-5.12/sage/env.py.orig sage-5.12/spkg/build/sage-5.12/sage/env.py
--- sage-5.12/spkg/build/sage-5.12/sage/env.py.orig	2013-08-12 15:14:58.586048479 -0300
+++ sage-5.12/spkg/build/sage-5.12/sage/env.py	2013-08-12 15:15:55.102050643 -0300
@@ -85,7 +85,7 @@ _add_variable_or_fallback('HOSTNAME',
 _add_variable_or_fallback('LOCAL_IDENTIFIER','$HOSTNAME.%s'%os.getpid())
         
 # bunch of sage directories and files
-_add_variable_or_fallback('SAGE_ROOT',       None)
+_add_variable_or_fallback('SAGE_ROOT',       '@@SAGE_ROOT@@')
 _add_variable_or_fallback('SAGE_LOCAL',      opj('$SAGE_ROOT', 'local'))
 _add_variable_or_fallback('SAGE_SHARE',      opj('$SAGE_LOCAL', 'share'))
         
@@ -95,7 +95,7 @@ _add_variable_or_fallback('SAGE_EXTCODE'
 _add_variable_or_fallback('SAGE_PACKAGES',   opj('$SAGE_ROOT', 'spkg'))
 _add_variable_or_fallback('SAGE_LOGS',       opj('$SAGE_ROOT', 'logs', 'pkgs'))
 _add_variable_or_fallback('SAGE_SPKG_INST',  opj('$SAGE_ROOT', 'spkg', 'installed'))
-_add_variable_or_fallback('SAGE_DOC',        opj('$SAGE_ROOT', 'devel', 'sage', 'doc'))
+_add_variable_or_fallback('SAGE_DOC',        '@@SAGE_DOC@@')
 _add_variable_or_fallback('DOT_SAGE',        opj(os.environ.get('HOME','$SAGE_ROOT'), '.sage'))
         
 # SAGE_LIB is the site-packages directory if the sage library
