diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2014-02-07 11:27:41.862680439 -0200
+++ src/module_list.py	2014-02-07 11:27:58.536681078 -0200
@@ -2069,7 +2069,8 @@ ext_modules = [
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([
