diff -up sage-5.12/spkg/build/sage-5.12/sage/libs/pari/decl.pxi.orig sage-5.12/spkg/build/sage-5.12/sage/libs/pari/decl.pxi
--- sage-5.12/spkg/build/sage-5.12/sage/libs/pari/decl.pxi.orig	2013-10-16 12:36:08.386258790 -0300
+++ sage-5.12/spkg/build/sage-5.12/sage/libs/pari/decl.pxi	2013-10-16 12:36:42.853260110 -0300
@@ -856,7 +856,6 @@ cdef extern from 'pari/pari.h':
     GEN     polredabs0(GEN x, long flag)
     GEN     polredabs2(GEN x)
     GEN     polredabsall(GEN x, long flun)
-    GEN     polredbest(GEN x, long flag)
     GEN     qflll0(GEN x, long flag)
     GEN     qflllgram0(GEN x, long flag)
     GEN     smallpolred(GEN x)
diff -up sage-5.12/spkg/build/sage-5.12/sage/libs/pari/gen.pyx.orig sage-5.12/spkg/build/sage-5.12/sage/libs/pari/gen.pyx
--- sage-5.12/spkg/build/sage-5.12/sage/libs/pari/gen.pyx.orig	2013-10-16 12:36:50.848260416 -0300
+++ sage-5.12/spkg/build/sage-5.12/sage/libs/pari/gen.pyx	2013-10-16 12:37:41.619262360 -0300
@@ -8081,10 +8081,6 @@ cdef class gen(sage.structure.element.Ri
         pari_catch_sig_on()
         return self.new_gen(polredabs0(self.g, flag))
 
-    def polredbest(self, flag=0):
-        pari_catch_sig_on()
-        return self.new_gen(polredbest(self.g, flag))
-
     def polresultant(self, y, var=-1, flag=0):
         t0GEN(y)
         pari_catch_sig_on()
