diff -up sage-5.10/spkg/build/sage-5.10/sage/libs/gap/util.pyx.orig sage-5.10/spkg/build/sage-5.10/sage/libs/gap/util.pyx
--- sage-5.10/spkg/build/sage-5.10/sage/libs/gap/util.pyx.orig	2013-05-06 14:24:03.160826749 -0300
+++ sage-5.10/spkg/build/sage-5.10/sage/libs/gap/util.pyx	2013-05-06 14:25:25.806829913 -0300
@@ -155,18 +155,9 @@ def gap_root():
        
         sage: from sage.libs.gap.util import gap_root
         sage: gap_root()   # random output
-        '/home/vbraun/opt/sage-5.3.rc0/local/gap/latest'
+        '/usr/lib/gap'
     """
-    import os.path
-    gapdir = os.path.join(SAGE_LOCAL, 'gap', 'latest')
-    if os.path.exists(gapdir):
-        return gapdir
-    print 'The gap-4.5.5.spkg (or later) seems to be not installed!'
-    gap_sh = open(os.path.join(SAGE_LOCAL, 'bin', 'gap')).read().splitlines()
-    gapdir = filter(lambda dir:dir.strip().startswith('GAP_DIR'), gap_sh)[0]
-    gapdir = gapdir.split('"')[1]
-    gapdir = gapdir.replace('$SAGE_LOCAL', SAGE_LOCAL)
-    return gapdir
+    return '/usr/lib/gap'
    
 
 cdef initialize():
