diff -up sage-5.13/spkg/build/sage-5.13/module_list.py.orig sage-5.13/spkg/build/sage-5.13/module_list.py
--- sage-5.13/spkg/build/sage-5.13/module_list.py.orig	2014-01-28 15:56:04.699175409 -0200
+++ sage-5.13/spkg/build/sage-5.13/module_list.py	2014-01-28 15:56:14.210175773 -0200
@@ -2052,7 +2052,8 @@ ext_modules = [
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([
