diff -up src/sage/all.py.orig src/sage/all.py
--- src/sage/all.py.orig	2016-08-08 13:29:42.192520823 -0400
+++ src/sage/all.py	2016-08-08 13:30:18.601522217 -0400
@@ -305,11 +305,11 @@ def _write_started_file():
 
     Check that the file exists when Sage is running::
 
-        sage: started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+        sage: started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
         sage: os.path.isfile(started_file)
         True
     """
-    started_file = os.path.join(SAGE_LOCAL, 'etc', 'sage-started.txt')
+    started_file = os.path.join(DOT_SAGE, 'sage-started.txt')
 
     # Current time with a resolution of 1 second
     import datetime
