diff -up src/module_list.py.orig src/module_list.py
--- src/module_list.py.orig	2016-08-08 13:44:56.193555823 -0400
+++ src/module_list.py	2016-08-08 13:45:10.146556357 -0400
@@ -563,11 +563,11 @@ ext_modules = [
               sources = ["sage/libs/ecl.pyx"],
               libraries = ["ecl"]),
 
-    Extension("sage.libs.fes",
-             ["sage/libs/fes.pyx"],
-             language = "c",
-             libraries = ['fes'],
-             package = 'fes'),
+#    Extension("sage.libs.fes",
+#             ["sage/libs/fes.pyx"],
+#             language = "c",
+#             libraries = ['fes'],
+#             package = 'fes'),
 
     Extension('sage.libs.flint.flint',
               sources = ["sage/libs/flint/flint.pyx"],
