diff -up sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima.py.orig sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima.py
--- sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima.py.orig	2012-08-04 13:21:54.100575138 -0400
+++ sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima.py	2012-08-04 13:22:15.580575956 -0400
@@ -544,7 +544,7 @@ class Maxima(MaximaAbstract, Expect):
         Expect.__init__(self,
                         name = 'maxima',
                         prompt = '\(\%i[0-9]+\)',
-                        command = 'maxima-noreadline --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP), 
+                        command = 'maxima --disable-readline --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP), 
                         maxread = 10000,
                         script_subdirectory = script_subdirectory,
                         restart_on_ctrlc = False,
@@ -588,7 +588,8 @@ class Maxima(MaximaAbstract, Expect):
 
         # Remove limit on the max heapsize (since otherwise it defaults
         # to 256MB with ECL).
-        self._sendline(":lisp (ext:set-limit 'ext:heap-size 0)")
+        self._sendline(":lisp #+ecl (ext:set-limit 'ext:heap-size 0) #-ecl 0")
+        self._sendline(':lisp #+gcl (progn (si:readline-off) (setf *error-output* (open "/dev/stderr" :direction :output)  *standard-input* (open "/dev/stdin" :direction :input)  *standard-output* (open "/dev/stdout" :direction :output))) #-gcl t')
         self._eval_line('0;')
 
     def __reduce__(self):
diff -up sage-5.5/spkg/build/sage_scripts-5.5/sage-maxima.lisp.orig sage-5.5/spkg/build/sage_scripts-5.5/sage-maxima.lisp
--- sage-5.5/spkg/build/sage_scripts-5.5/sage-maxima.lisp.orig	2012-08-04 13:22:07.842575668 -0400
+++ sage-5.5/spkg/build/sage_scripts-5.5/sage-maxima.lisp	2012-08-04 13:22:15.580575956 -0400
@@ -3,4 +3,8 @@
 ;(setf *general-display-prefix* "<sage-display>")
 
 (setf *prompt-prefix* "<sage-display>")
-
+#+clisp
+(setf
+  *error-output* (open "/dev/stderr" :direction :output)
+  *standard-input* (open "/dev/stdin" :direction :input)
+  *standard-output* (open "/dev/stdout" :direction :output))
