diff -up src/sage/sandpiles/sandpile.py.orig src/sage/sandpiles/sandpile.py
--- src/sage/sandpiles/sandpile.py.orig	2016-08-08 13:33:09.209528750 -0400
+++ src/sage/sandpiles/sandpile.py	2016-08-08 13:33:36.251529786 -0400
@@ -346,9 +346,6 @@ from sage.arith.all import falling_facto
 from sage.rings.all import Integer, PolynomialRing, QQ, ZZ
 from sage.symbolic.all import I, pi
 
-# TODO: remove the following line once 4ti2 functions are removed
-path_to_zsolve = os.path.join(SAGE_LOCAL,'bin','zsolve')
-
 class Sandpile(DiGraph):
     """
     Class for Dhar's abelian sandpile model.
@@ -5229,7 +5226,7 @@ class SandpileDivisor(dict):
         sign_file.close()
         # compute
         try:
-            os.system(path_to_zsolve+' -q ' + lin_sys + ' > ' + lin_sys_log)
+            os.system('zsolve -q ' + lin_sys + ' > ' + lin_sys_log)
             # process the results
             zhom_file = open(lin_sys_zhom,'r')
         except IOError:
