diff -up sage-5.12/spkg/build/sage-5.12/module_list.py.orig sage-5.12/spkg/build/sage-5.12/module_list.py
--- sage-5.12/spkg/build/sage-5.12/module_list.py.orig	2013-10-16 11:54:03.841162116 -0300
+++ sage-5.12/spkg/build/sage-5.12/module_list.py	2013-10-16 11:54:14.330162518 -0300
@@ -2029,7 +2029,8 @@ ext_modules = [
 # These extensions are to be compiled only if the
 # corresponding packages have been installed
 
-from sage.misc.package import is_package_installed
+def is_package_installed(name):
+    return False
 
 if is_package_installed('fes'):
     ext_modules.extend([
