diff -up src/sage/groups/perm_gps/permgroup.py.orig src/sage/groups/perm_gps/permgroup.py
--- src/sage/groups/perm_gps/permgroup.py.orig	2016-08-20 15:03:27.119562623 -0400
+++ src/sage/groups/perm_gps/permgroup.py	2016-08-20 15:05:17.650566855 -0400
@@ -165,6 +165,10 @@ def load_hap():
         gap.load_package("hap")
     except Exception:
         gap.load_package("hap")
+        try:
+            gap.load_package("hap")
+        except Exception:
+            raise RuntimeError("gap-hap package failed to load.")
 
 def hap_decorator(f):
     """
@@ -190,8 +194,6 @@ def hap_decorator(f):
     """
     @wraps(f)
     def wrapped(self, n, p=0):
-        if not is_package_installed('gap_packages'):
-            raise RuntimeError("You must install the optional gap_packages package.")
         load_hap()
         from sage.arith.all import is_prime
         if not (p == 0 or is_prime(p)):
