diff -up src/bin/sage-maxima.lisp.orig src/bin/sage-maxima.lisp
--- src/bin/sage-maxima.lisp.orig	2016-08-08 13:31:41.528525393 -0400
+++ src/bin/sage-maxima.lisp	2016-08-08 13:31:54.865525903 -0400
@@ -4,3 +4,8 @@
 
 (setf *prompt-prefix* "<sage-display>")
 
+#+clisp
+(setf
+  *error-output* (open "/dev/stderr" :direction :output)
+  *standard-input* (open "/dev/stdin" :direction :input)
+  *standard-output* (open "/dev/stdout" :direction :output))
diff -up src/sage/interfaces/maxima.py.orig src/sage/interfaces/maxima.py
--- src/sage/interfaces/maxima.py.orig	2016-08-08 13:32:02.623526200 -0400
+++ src/sage/interfaces/maxima.py	2016-08-08 13:32:41.698527697 -0400
@@ -556,7 +556,7 @@ class Maxima(MaximaAbstract, Expect):
         Expect.__init__(self,
                         name = 'maxima',
                         prompt = '\(\%i[0-9]+\) ',
-                        command = 'maxima --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP),
+                        command = 'maxima --disable-readline --userdir="%s" -p "%s"'%(SAGE_MAXIMA_DIR,STARTUP), 
                         script_subdirectory = script_subdirectory,
                         restart_on_ctrlc = False,
                         verbose_start = False,
@@ -623,7 +623,8 @@ class Maxima(MaximaAbstract, Expect):
 
         # Remove limit on the max heapsize (since otherwise it defaults
         # to 256MB with ECL).
-        self._sendline(":lisp (ext:set-limit 'ext:heap-size 0)")
+        self._sendline(":lisp #+ecl (ext:set-limit 'ext:heap-size 0) #-ecl 0")
+        self._sendline(':lisp #+gcl (progn (si:readline-off) (setf *error-output* (open "/dev/stderr" :direction :output)  *standard-input* (open "/dev/stdin" :direction :input)  *standard-output* (open "/dev/stdout" :direction :output))) #-gcl t')
         self._eval_line('0;')
 
         # set random seed
