diff -up sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima_lib.py.orig sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima_lib.py
--- sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima_lib.py.orig	2012-12-29 15:42:54.811218546 -0200
+++ sage-5.5/spkg/build/sage-5.5/sage/interfaces/maxima_lib.py	2012-12-29 15:43:06.739219003 -0200
@@ -77,7 +77,7 @@ from maxima_abstract import (MaximaAbstr
 ## We begin here by initializing Maxima in library mode
 ## i.e. loading it into ECL
 ecl_eval("(setf *load-verbose* NIL)")
-ecl_eval("(require 'maxima)")
+ecl_eval("(require 'maxima.system)")
 ecl_eval("(in-package :maxima)")
 ecl_eval("(setq $nolabels t))")
 ecl_eval("(defvar *MAXIMA-LANG-SUBDIR* NIL)")
