diff -up src/sage/numerical/backends/coin_backend.pyx.orig src/sage/numerical/backends/coin_backend.pyx
--- src/sage/numerical/backends/coin_backend.pyx.orig	2016-08-08 13:41:10.063547164 -0400
+++ src/sage/numerical/backends/coin_backend.pyx	2016-08-08 13:41:28.426547867 -0400
@@ -755,8 +755,8 @@ cdef class CoinBackend(GenericBackend):
         model.setLogLevel(old_logLevel)
 
         # multithreading
-        import multiprocessing
-        model.setNumberThreads(multiprocessing.cpu_count())
+        #import multiprocessing
+        #model.setNumberThreads(multiprocessing.cpu_count())
 
         model.branchAndBound()
 
